### DISCUSSION TITLE
NextJS Access Logs

### DISCUSSION THREAD

**Discussion by jamesbrooks94** [vercel/next.js#56658] [App Router]
I'm struggling to create access logs for my Next app using App Router where log lines are emitted after a request has finished. For example, I'd like to get an access log for each of my API endpoints, all pages (all of mine are dynamic so are server side rendered).

The only way I can seem to get an access log is by creating a custom server. Additionally, I can't seem to get a span started within the custom server to propagate through to the request handlers.

Ideal access log format example: `GET /api/hello 200 2ms`

If there is a solution for this, I haven't been able to find it after hours of searching. Could we either add this as a feature or if available update the docs.

**Comment by x-yuri** [comment]
[Custom server](https://nextjs.org/docs/pages/building-your-application/configuring/custom-server), reverse proxy, or... [patch `nextjs`](https://stackoverflow.com/a/78067195/52499) :)
